CONTRIBUTING.md

```diff
@@ -0,0 +1,67 @@
+# Contribution Guidelines
+
+This document aims to provide guidelines for merging changesets into the sparrow master branch.
+
+For the purpose of this document, we classify changesets/pull requests as:
+
+* big: major changes to core code
+* non-trivial: significant changes, fixes and improvements. Especially if they change core modules.
+* trivial: minor bugfixes, small changes to auxiliary modules, changes to documentation, and patches which change only test code.
+
+## Writing & Testing Code:
+
+Write type specifications and function signatures because they're remarkably helpful when it comes to reading the source.
+Use `mix format` for formatting the code.
+Test newly added code. Ensure test coveridge is not decreased. If adding new module `<my_new_module_name>.ex` add also `<my_new_module_name>_test.exs` in test directory in corresponding subdirectory.
+
+What makes a good comment?
+Write about why something is done a certain way.
+E.g. explain a why a decision was made or describe a subtle but tricky case.
+We can read a test case or the source, respectively, to see **what** the code does or **how** it does it.
+Comments should give us more insight into **why** something was done (the reasoning).
+
+## 1. Preparation
+
+### Branch and code
+
+Always create a branch with a descriptive name from an up-to-date master.
+Do your work in the branch, push it to the ESL repository if you have access to it, otherwise to your own repository.
+
+### Run tests
+
+When done, run `mix test` and write tests related to what you've done.
+
+### Check coding style
+
+Use `mix format` for formatting the code.
+
+### Push
+
+Then push the changes and create a pull request to master, following the PR description template.
+Make sure all Travis tests pass (if only some jobs fail it is advisable to restart them, since they sometimes
+fail at random).
+
+## 2. Review
+
+Both trivial and non-trivial PRs have to be reviewed by at least one other person from the core development team.
+For big changesets consult the Tech Lead.
+The reviewer's remarks should be placed as inline comments on github for future reference.
+
+Then, apply the reviewer's suggestions.
+If the changes are limited to documentation or code formatting, remember to prefix commit message with "[skip ci]" so tests results are not a concern.
```

Comment:
If the changes are limited to documentation or code formatting, remember to prefix commit message with "[skip ci]" so that the tests results are not a concern.